tools/clang/lib/CodeGen/CGStmt.cpp

```diff
@@ -1207,6 +1207,7 @@ void CodeGenFunction::EmitBreakStmt(const BreakStmt &S) {
     assert(EHStack.getInnermostActiveNormalCleanup() == EHStack.stable_end() && "HLSL Shouldn't need cleanups");
     CGM.getHLSLRuntime().EmitHLSLCondBreak(*this, CurFn, BreakContinueStack.back().BreakBlock.getBlock(),
                                            BreakContinueStack.back().ContinueBlock.getBlock());
+    Builder.ClearInsertionPoint();
```

Comment:
Consider this line: https://github.com/microsoft/DirectXShaderCompiler/blob/7e0f44fa6f56365eb1148ef5a394e1ce35c92343/tools/clang/lib/CodeGen/CGCleanup.cpp#L964 which was the portion of break generation that was missed here.

Reply:
This is where that setting is checked: https://github.com/microsoft/DirectXShaderCompiler/blob/7e0f44fa6f56365eb1148ef5a394e1ce35c92343/tools/clang/lib/CodeGen/CGStmt.cpp#L54